doc/dev/ceph_krb_auth.rst

```diff
@@ -1,335 +1,545 @@
+        the principal is in the default realm for the context in which it is being used.
+
+        + Usually, a principal is divided into three parts:
+            * the ``primary``, the ``instance``, and the ``realm``
+
+            * The format of a typical Kerberos V5 principal is ``primary/instance@REALM``.
+
+            * The ``primary`` is the first part of the principal. In the case of a user,
+            it's the same as the ``username``. For a host, the primary is the word ``host``.
+            For Ceph, will use ``ceph`` as a primary name which makes it easier to organize
+            and identify Ceph related principals.
+
+            * The ``instance`` is an optional string that qualifies the primary. The instance
+            is separated from the primary by a slash ``/``. In the case of a user, the
+            instance is usually ``null``, but a user might also have an additional principal,
+            with an instance called ``admin``, which he/she uses to administrate a database.
+            The principal ``sage@CEPH.COM`` is completely separate from the principal
+            ``sage/admin@CEPH.COM``, with a separate password, and separate permissions.
+            In the case of a host, the instance is the fully qualified hostname,
+            i.e., ``osd1.ceph.com``.
+
+            * The ``realm`` is the Kerberos realm. Usually, the Kerberos realm is the domain
+            name, in *upper-case letters*. For example, the machine ``osd1.ceph.com``
+            would be in the realm ``CEPH.COM``.
+
+    - Keytab
+        + A keytab file stores the actual encryption key that can be used in lieu of a
+        password challenge for a given principal. Creating keytab files are useful for
+        noninteractive principals, such as *Service Principal Names*, which are often
+        associated with long-running processes like Ceph daemons. A keytab file does not
+        have to be a "1:1 mapping" to a single principal. Multiple different principal keys
+        can be stored in a single keytab file.
+
+            * The keytab file allows a user/service to authenticate without knowledge of
+            the password. Due to this, *keytabs should be protected* with appropriate
+            controls to prevent unauthorized users from authenticating with it.
+
+            * The default client keytab file is ``/etc/krb5.keytab``
+
+
+The 'Ceph side' of the things
+------------------------------
+
+In order to configure connections (from Ceph nodes) to the KDC:
+
+1. Login to the Kerberos client (Ceph server nodes) and confirm it is properly configured,
+by checking and editing ``/etc/krb5.conf`` file properly.
+::
+    /etc/krb5.conf
+    [libdefaults]
+        dns_canonicalize_hostname = false
+        rdns = false
+        forwardable = true
+        dns_lookup_realm = true
+        dns_lookup_kdc = true
+        allow_weak_crypto = false
+        default_realm = CEPH.COM
```

Comment:
Just wondering: does the realm need to be in all-caps? Is there a reason for it to be in all-caps if not?

If there's no reason for the realm to be in all caps, could we please not reinforce the CEPH stuff in our docs, and use instead `Ceph` (if it really has to be), or just `ceph`? Alternatively, if it *has* to be all-caps, could we change this for a placeholder instead? Like, "DOMAIN" or something?

Reply:
Well, in theory, it doesn't have to, however, that is a standard convention for Kerberos Domain/Realm (it is not meant to be used as `Ceph`). *That's how we differentiate problems with the Kerberos service from problems with the DNS namespace while keeping a name that is familiar*. That's also mentioned here: https://github.com/ceph/ceph/pull/23550/commits/5eda2deada5579b9ab7c1318c4a09dc9755f864a#diff-d679fd9648baf82cfd9bc8bc91ea1a8eR285 
I've replaced `CEPH.COM` and `ceph.com` to `MYDOMAIN.COM` and `mydomain.com` respectively.